subsystem/metrics_test.go

```diff
@@ -44,12 +45,26 @@ func isHostValidationInFailedStatus(clusterID, hostID strfmt.UUID, validationID
 			if v.ID != validationID {
 				continue
 			}
-			return v.Status == "failure", nil
+			return v.Status == expectedStatus, nil
 		}
 	}
 	return false, nil
 }
 
+func waitForHostValidationStatus(clusterID, hostID strfmt.UUID, expectedStatus string, hostValidationIDs ...models.HostValidationID) {
+
+	waitFunc := func() (bool, error) {
+		res := true
+		for _, vID := range hostValidationIDs {
+			cond, _ := isHostValidationInStatus(clusterID, hostID, vID, expectedStatus)
+			res = res && cond
+		}
+		return res, nil
+	}
```

Comment:
```suggestion
	waitFunc := func() (bool, error) {
		for _, vID := range hostValidationIDs {
			cond, _ := isHostValidationInStatus(clusterID, hostID, vID, expectedStatus)
			if !cond{
			    return false, fmt.Errorf("Host validation failed: cluster id: %s, host id: %s, validation id: %s, expected status: %s,", clusterID, hostID, vID, expectedStatus)
			}
		}
		return true, nil
	}
```

Reply:
Partially fixed, returning an error will stop the waiting loop, we don't want that - instead we want to continue looping until it is true/timeout.